src/main/java/edu/kit/provideq/toolbox/sat/solvers/GamsSatSolver.java

```diff
@@ -59,7 +57,7 @@ public void solve(Problem<String> problem, Solution<DimacsCnfSolution> solution,
             GamsProcessRunner.class,
             satPath,
             "sat.gms")
-        .run(problem.type(), solution.getId(), dimacsCnf.toString());
+        .run(getProblemType(), solution.getId(), dimacsCnf.toString());
```

Comment:
The change from a problem type enum to a class with generics creates a bit of a problem in the ProcessRunners as they now have raw use of the `ProblemType` class.

Reply:
Oh I must have missed those... I tried to fix all of these usages but unfortunately, the java compiler and checkstyle don't report "raw usage" warnings. Fixed this for process runners in 774eb55

Reply:
I was able to spot these via the IDE's code inspector (IntelliJ)